www/assets/js/spreadsheet.js

```diff
@@ -418,13 +418,40 @@ const spreadsheet = (function() {
     }
 
     function defaultViewer(rect,offset,el) {
+      const spreadsheetElement = document.getElementById("slo-spreadsheet");
+      const spreadsheetSize = spreadsheetElement.getBoundingClientRect();
+      const boxWidth = spreadsheetSize.width/1.618 // golden ratio
+      const iconSize = 60
+      const standardBoxWidth = 300 + iconSize
+      const extraPadding = 30
+
       let columnDef = getColumnDefinition(el)
       let row = getRow(el)
       selector.innerHTML = ''
       selector.style.top = Math.max(2, (rect.top - offset.top))+'px'
       selector.style.left = (rect.left - offset.left)+'px'
-      selector.style.width = rect.width+'px'
+      selector.style.width = Math.max(300, (rect.width))+'px'
```

Comment:
Do Math.min(spreadsheetSize.width, Math.max(300, rect.width)+'px'
or Math.clamp if that exists